### PR TITLE
DevExtreme now uses Algorithm 1 specified object structure

### PR DESCRIPTION
**Pull request by cbuchko**
Removed the mock data we were using before, the CalendarView now uses mock data provided by the algorithm 1 team. This means theoretically that once the backend is actually sending us data, it should work if its passed in as a prop to the CalendarView component.

This PR does not include any integration with algorithm 2 data, that will be a separate PR once we know more about their objects structure.

### CODE REVIEW

File: src/components/Schedule/CalendarView.tsx [hunk 1]
```diff
@@ -72,43 +17,85 @@ const getWeekDay = (day: Date) => {
...
 
-export const CalendarView = () => {
+//convert Course object data from backend to the structure DevExtreme expects
+const buildAppointments = (courses: Course[]) => {
+  const appointments = courses
+    .map((course) => {
+      const assignedCourses = splitCourseDays(course);
+      return assignedCourses;
+    })
+    .flat();
```
Comment (jayden-chan) [inline src/components/Schedule/CalendarView.tsx]: Nitpick, but you can just use `flatMap` here

File: src/stores/preferences.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    fallTermCourses: number;
+    springTermCourses: number;
+    summerTermCourses: number;
+  };
+  
+  export type Preference = {
+    course: Course;
+    preferenceNum: number;
+  };
+
```
Comment (jayden-chan) [inline src/stores/preferences.ts]: Pretty bizarre indentation here, are you using Prettier? I'm going to keep nagging about this until the code looks pretty :stuck_out_tongue:

**Review by jayden-chan — DISMISSED** [review]

File: src/stores/preferences.ts (hunk 2 above)
Comment (cbuchko) [reply src/stores/preferences.ts]: LOL I am but I forgot on this file sad :(

File: src/components/Schedule/CalendarView.tsx (hunk 1 above)
Comment (cbuchko) [reply src/components/Schedule/CalendarView.tsx]: oh  that's nice, I didn't even know that existed lol

File: src/stores/preferences.ts (hunk 2 above)
Comment (cbuchko) [reply src/stores/preferences.ts]: updated

Comment (jayden-chan) [reply src/stores/preferences.ts]: You should configure it to format the file when you save. Not only will this keep everything formatted all the time, but you can copy/paste around severely mal-formatted code and everything will be fixed for you. Saves a ton of time.

**Review by jayden-chan — APPROVED** [review]

File: src/stores/preferences.ts (hunk 2 above)
Comment (cbuchko) [reply src/stores/preferences.ts]: yeah that's a good idea haha

**Review by nicudy — APPROVED** [review]
Looks good!

**Pull request merged**